services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,37 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  xQueueSend(thermalMgrQueueHandle, event, portMAX_DELAY);
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t interruptEvent;
+  interruptEvent.type = THERMAL_MGR_EVENT_OS_INTERRUPT;
+  thermalMgrSendEvent(&interruptEvent); // Sends interrupt event for MGR to see if too high or low
+
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
   while (1) {
-    
+    thermal_mgr_event_t event;
+     if (xQueueReceive(thermalMgrQueueHandle, &event, portMAX_DELAY) == pdTRUE){
+      float curTemp;
+      float interruptTemp;
+        switch (event.type) {
+          case THERMAL_MGR_EVENT_MEASURE_TEMP_CMD:
+            readTempLM75BD(LM75BD_OBC_I2C_ADDR, &curTemp);
```

Comment:
handle the case where readTempLM75BD does not return ERR_CODE_SUCCESS (log the error code).